### PR TITLE
Updated PR from anurse

### PR DESCRIPTION
**Pull request by HaoK**
Rebased and updated version of https://github.com/aspnet/Docs/pull/1431

It wouldn't let me rebase in andrew's branch so I had to create a new branch/PR

@Rick-Anderson @anurse @blowdart @rachelappel

### CODE REVIEW

File: aspnet/security/authorization/dependencyinjection.rst [hunk 1]
```diff
@@ -18,7 +18,7 @@ For example, if you wanted to use ASP.NET's logging infrastructure you would to
          _logger = loggerFactory.CreateLogger(this.GetType().FullName);
      }
 
-     protected override void Handle(AuthorizationContext context, MyRequirement requirement)
+     protected override void Handle(AuthorizationHandlerContext context, MyRequirement requirement)
```
Comment (blowdart) [inline aspnet/security/authorization/dependencyinjection.rst]: async?

**Comment by Rick-Anderson** [comment]
@HaoK  I don't see any security changes that need @blowdart  so I'll merge this.

**Comment by Rick-Anderson** [comment]
@HaoK  I'll fix the build problems.

**Comment by Rick-Anderson** [comment]
@HaoK  can you address blowdart async comment? Can you squash and get rid of merge conflicts? Once dan updates the API ref, I'll put the API links back in.

File: aspnet/security/authorization/dependencyinjection.rst (hunk 1 above)
Comment (HaoK) [inline aspnet/security/authorization/dependencyinjection.rst]: This was fixed in another PR already https://github.com/aspnet/Docs/pull/1454

**Comment by HaoK** [comment]
Replaced by https://github.com/aspnet/Docs/issues/1467

**Pull request closed without merging**